From: Steve Langasek <steve.langasek@canonical.com>
Date: Thu, 3 Oct 2013 00:00:00 +0200
Subject: Ignore certificate expiries when verifying signatures

The UEFI implementation explicitly ignores all errors due to expired
(or not yet valid) signatures.  Ensure that sbverify behaves compatibly.

Bug-Ubuntu: https://bugs.launchpad.net/bugs/1234649.
---
 src/sbverify.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/sbverify.c b/src/sbverify.c
index fb03d21..0d67fad 100644
--- a/src/sbverify.c
+++ b/src/sbverify.c
@@ -206,6 +206,13 @@ static int x509_verify_cb(int status, X509_STORE_CTX *ctx)
 		if (cert_in_store(ctx->current_cert, ctx))
 			status = 1;
 	}
+	/* UEFI doesn't care about expired signatures, so we shouldn't either. */
+	else if (err == X509_V_ERR_CERT_HAS_EXPIRED ||
+			err == X509_V_ERR_ERROR_IN_CERT_NOT_BEFORE_FIELD ||
+			err == X509_V_ERR_CERT_NOT_YET_VALID ||
+			err == X509_V_ERR_ERROR_IN_CERT_NOT_AFTER_FIELD) {
+		status = 1;
+	}
 
 	return status;
 }
